### PR TITLE
Some doc work

### PR DESCRIPTION
**Pull request by grebe**
I'm reorganizing and cleaning up the documentation. I'm also adding documentation for the stuff in rocket.

### CODE REVIEW

File: doc/Blackbox Compatibility.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+> Note: These are only good for golden model testing. They should *never* be used in your final design. 
+
+* SInt -> Real
+* Real -> SInt
+
+## Non-synthesizable operations that don't have Chisel number type equivalents (with Verilator support)
+
+> Note: These work with Verilator + FIRRTL interpreter, but can't be used with type classes.
+
```
Comment (chick) [inline doc/Blackbox Compatibility.md]:
```suggestion
> Note: These work with Verilator or Treadle, but can't be used with type classes.

```

File: doc/Example.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+// Scala unit testing style
+import org.scalatest.{FlatSpec, Matchers}
+
+// IO Bundle. Note that when you parameterize the bundle, you may need to override cloneType.
+// This also creates x, y, z inputs/outputs (direction must be specified at some IO hierarchy level)
+// of the type you specify via gen (must be Data:RealBits = UInt, SInt, FixedPoint, DspReal)
+class SimpleDspIo[T <: Data:RealBits](gen: T) extends Bundle {
+  val x = Input(gen.cloneType)
+  val y = Input(gen.cloneType)
+  val z = Output(gen.cloneType)
```
Comment (chick) [inline doc/Example.md]: I'm not completely sure but I think Input and Output will call clonetype for you
```suggestion
  val x = Input(gen)
  val y = Input(gen)
  val z = Output(gen)
```

File: doc/Testing.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
+# Details on DspTesterOptionsManager
+* As in the example above, the DspTesterOptionsManager manages the following category of options listed hierarchically [objectName = caseClass] (Case class files linked):
+  * dspTesterOptions = [DspTesterOptions()](https://github.com/ucb-bar/dsptools/blob/add_ops/src/main/scala/dsptools/tester/DspTesterOptions.scala)
+  * testerOptions = [TesterOptions()](https://github.com/ucb-bar/chisel-testers/blob/master/src/main/scala/chisel3/iotesters/TesterOptions.scala)
+  * interpreterOptions = [InterpreterOptions()](https://github.com/ucb-bar/firrtl-interpreter/blob/master/src/main/scala/firrtl_interpreter/Driver.scala)
```
Comment (chick) [inline doc/Testing.md]: Add treadle options
```suggestion
 * interpreterOptions = [InterpreterOptions()](https://github.com/ucb-bar/firrtl-interpreter/blob/master/src/main/scala/firrtl_interpreter/Driver.scala)
  * treadleOptions = [TreadleOptions()](https://github.com/freechipsproject/treadle/blob/master/src/main/scala/treadle/Driver.scala)
```

File: doc/Example.md (hunk 2 above)
Comment (grebe) [reply doc/Example.md]: I think you're right

**Pull request merged**